src/d3a/d3a_core/export.py

```diff
@@ -241,6 +255,51 @@ def _export_area_stats_csv_file(self, area: Area, directory: dir, balancing: boo
         except Exception as ex:
             _log.error("Could not export area data: %s" % str(ex))
 
+    def plot_device_stats(self, area: Area, node_address_list: list):
+        """
+        Wrapper for _plot_trade_partner_cell_tower
+        """
+        new_node_address_list = node_address_list + [area.name]
+        for child in area.children:
+            if child.children:
+                self.plot_device_stats(child, new_node_address_list)
+            else:
+                address_list = new_node_address_list + [child.name]
+                self._plot_device_stats(address_list)
+
+    def _plot_device_stats(self, address_list: list):
+        """
+        Plots device graphs
+        """
+        # Dont use the root area name for address list:
+        device_address_list = address_list[1::]
```

Comment:
From offline discussion with @faizan2590 , this should not be handled like this. There should always be a "brig" member in `self.endpoint_buffer.device_statistics.device_stats_dict` even though there is not device directly connected to it. Will change this in `device_statistics.py`.